hash/hash_test.go

```diff
@@ -37,3 +37,31 @@ func TestGenMerkleHashFunc(t *testing.T) {
 	// different children (e.g. different order) -> different hash
 	r.NotEqual(GenMerkleHashFunc(aChallenge)(nil, lChild, rChild), GenMerkleHashFunc(aChallenge)(nil, rChild, lChild))
 }
+
+func TestGenLabelHashFuncHash(t *testing.T) {
+	challenge := []byte("a123123123mskofvuw098e71bnj91273")
+	data := []byte("12312390819023879012379184718920408data0123128991231239081")
+
+	require.Equal(
+		t,
+		[]byte{
+			104, 1, 230, 180, 3, 202, 139, 10, 164, 69, 74, 169, 116, 77, 27, 134,
+			138, 163, 204, 32, 182, 197, 205, 194, 13, 48, 30, 224, 254, 1, 46, 133,
+		},
+		GenLabelHashFunc(challenge)(data),
+	)
+}
```

Comment:
Depending on the system these tests are executed they only test one of the two implementations.

You could extract the 2 cases of the `switch` in `GenLabelHashFunc` as private methods, call them explicitly in one test and compare if their output is the same. Similarly for `GenMerkleHashFunc` below.

Reply:
Yes, but anyway it's impossible to test the optimized version if host doesn't support it. It would:
- test both if host supports it
- test only generic if host doesn't support it.

Reply:
The important part imo that we have a test that compares the output of both implementations and makes sure they are the same.

Reply:
Done